### PR TITLE
logic for data_transform

### PR DESCRIPTION
**Pull request by codemeasandwich**
In `preprocess.py`:
   - Added the `translate_to_en` function to translate non-English text to English using the `stanza` and `transformers` libraries.

In `data_transform`:
   - Added the method to handle the case when the input data is not a DataFrame.
   - If the input data is not a DataFrame, it assigns the input data directly to `self.embeddings`, assuming it's already transformed before applying the preprocessing pipeline.
   - Performs label encoding on the target variable.

In `model/base.py`:
   - Added a `data_transform` method to perform data transformation on the input data.
   - If the input data is a DataFrame, it applies a `ColumnTransformer` with `StandardScaler` for numeric features and `OneHotEncoder` for categorical features.
   - If the input data is not a DataFrame, it assumes it's already transformed and assigns it directly to `self.embeddings`.
   - Converts the target variable to integers if needed.
   - Added a `build` method to update the model's attributes with provided values.

These changes address the missing code and provide appropriate implementations based on the context of the application. The `translate_to_en` function is added to handle translation of non-English text, and the `data_transform` methods are updated to handle data preprocessing consistently across different models.

**Note that the code assumes the required libraries (`stanza`, `transformers`, `sklearn`, etc.) are installed and available in the environment.**

### CODE REVIEW

File: modelling/modelling.py [hunk 1]
```diff
@@ -44,7 +44,57 @@ def model_predict(data, df, name):
     model.train(data)
     model.predict(data.X_test)
     model.print_results(data)
-
-
+    
+    return results
+def model_predict_chained(data, df, name):
```
Comment (dimitrios-vasileiou-kalfas) [inline modelling/modelling.py:49]: This is what is asked but is not called anywhere

File: modelling/modelling.py [hunk 2]
```diff
@@ -44,7 +44,57 @@ def model_predict(data, df, name):
...
+    model.print_results(data)
+    
+    model = RandomForest("RandomForest", data.get_embeddings(), data.get_y2_y3_y4())
+    model.train(data)
+    model.predict(data.X_test)
+    res = model.print_results(data)
+    results.append(res)
+    
+    return results
+def model_predict_hierarchical(data, df, name):
```
Comment (dimitrios-vasileiou-kalfas) [inline modelling/modelling.py:70]: This was not asked to be implemented  (Decision 2)

**Review by dimitrios-vasileiou-kalfas — CHANGES_REQUESTED** [review]
put some comments there that i dont understand

File: modelling/modelling.py (hunk 2 above)
Comment (codemeasandwich) [reply modelling/modelling.py:70]: The task clearly asks to consider and implement the Hierarchical Modeling approach (Design Decision 2) in addition to the Chained Multi-outputs approach (Design Decision 1).

The `model_predict_hierarchical` function starting at line 70 is indeed implementing the Hierarchical Modeling approach as described in the task. Specifically:

1. I first train a model to predict Type 2 classes.
2. For each Type 2 class, I filter the data and train a new model to predict Type 3 classes within that Type 2 class.
3. For each Type 2 and Type 3 class combination, I again filter the data and train models to predict Type 4 classes.

This aligns with the description in the task:

> For example, a model instance (of e.g., random forest) will classify Type 2, then data from Type 3 will be filtered for each class in Type 2 and a new model instance will perform classification on that filtered data. [...] Then, at the next level, we consider each filter set A as a data set for Type 3 and will filter Type 4 under each filter similarly set A as we did for Type 2-Type 3 combination.

**Comment by codemeasandwich** [comment]
@dimitrios-vasileiou-kalfas 
> put some comments there that i dont understand

What "comments" do you not understand?

File: modelling/modelling.py (hunk 2 above)
Comment (dimitrios-vasileiou-kalfas) [reply modelling/modelling.py:70]: Apologies @codemeasandwich I still i dont see where it asks to implement Decision Choice 2
Screenshot 2024-04-22 at 09 13 53
